doc/src/guides/solving/solve-ode.md

```diff
@@ -0,0 +1,393 @@
+>>> solutions_list = [{fn.lhs:fn.rhs for fn in solution} for solution in solutions]
+>>> solutions_list # nested comprehension approach
+[{y(x): C1 - C2*exp(x), z(x): C2*exp(x)}, {y(x): C1 + C2*exp(x), z(x): C2*exp(x)}]
+>>> solutions_list[0][y(x)]
+C1 - C2*exp(x)
+```
+
+### Work With Arbitrary Constants
+
+You can manipulate arbitrary constants such as `C1`, `C2`, and `C3`, which are
+generated automatically by {func}`~.dsolve`, by creating them as symbols. For
+example, if you want to assign values to arbitrary constants, you can create
+them as symbols and then substitute in their values using
+{meth}`~sympy.core.basic.Basic.subs`:
+
+```py
+>>> from sympy import Function, dsolve, Derivative, symbols, pi
+>>> x, C1, C2 = symbols("x, C1, C2")
+>>> y = Function('y')
+>>> result = dsolve(Derivative(y(x), x, x) + 9*y(x), y(x)).rhs
+>>> result
+C1*sin(3*x) + C2*cos(3*x)
+>>> result.subs({C1: 7, C2: pi})
+7*sin(3*x) + pi*cos(3*x)
+```
+
+### Numerically Solve an ODE in SciPy
+
+A common workflow which leverages
+[SciPy's](https://docs.scipy.org/doc/scipy/index.html) fast numerical ODE
+solving is
+1. set up an ODE in SymPy
+2. convert it to a lambda function using {func}`~.lambdify`
+3. solve it numerically using SciPy's `solve_ivp`.
+
+```{warning}
+{func}`~.lambdify` uses {external:func}`~.exec` to dynamically execute Python code, and thus should not be used on unsanitized input.
+```
+
+Here is an [example from the field of chemical
+kinetics](https://www.sympy.org/scipy-2017-codegen-tutorial/notebooks/25-chemical-kinetics-intro.html):
+
+```py
+>>> from sympy import symbols, lambdify
+>>> import numpy as np
+>>> import scipy.integrate
+>>> def rhs(t, y, kf, kb):
+...     rf = kf * y[0]**2 * y[1]
+...     rb = kb * y[2]**2
+...     return [2*(rb - rf), rb - rf, 2*(rf - rb)]
+>>> y, (kf, kb) = symbols('y:3'), symbols('kf kb')
+>>> ydot = rhs(None, y, kf, kb)
+>>> t = symbols('t') # not used in this case
+>>> f = lambdify((t, y, kf, kb), ydot)
+>>> k_vals = np.array([0.42, 0.17]) # arbitrary in this case
+>>> y0 = [1, 1, 0]
+>>> scipy.integrate.solve_ivp(f, (0, 10), y0, args=k_vals)
+    {'message': 'The solver successfully reached the end of the integration interval.', 'nfev': 68, 'njev': 0, 'nlu': 0, 'sol': None, 'status': 0, 'success': True, 't': [0.00000000e+00 1.68190462e-03 1.85009508e-02 1.86691413e-01
+     6.37253319e-01 1.27438822e+00 2.15637690e+00 3.28555351e+00
+     4.69240977e+00 6.45455786e+00 8.73068099e+00 1.00000000e+01], 't_events': None, 'y': [[1.         0.99858969 0.98475531 0.86889862 0.68120238 0.55390608
+      0.47951256 0.44569558 0.43354565 0.43020361 0.42955182 0.4294468 ]
+     [1.         0.99929485 0.99237766 0.93444931 0.84060119 0.77695304
+      0.73975628 0.72284779 0.71677282 0.71510181 0.71477591 0.7147234 ]
+     [0.         0.00141031 0.01524469 0.13110138 0.31879762 0.44609392
+      0.52048744 0.55430442 0.56645435 0.56979639 0.57044818 0.5705532 ]], 'y_events': None}
+```
```

Comment:
We have the matplolib extension installed so you can create graphs. This data is more easily understood if you make a simple plot. See the beam module docs for examples of using the plotting extension (or the plotting module).

Reply:
Thanks. I made a graph which displays properly in a Jupyter Notebook. But I'm having trouble getting the plot to display properly via Sphinx. I was able to get a simple matplotlib example to display, but not the ODE solution. Also, may not be relevant once the ODE plot works, but if I make a second simple plot with different values, it shows the first plot again:

<img width="530" alt="matplotlib for ODE" src="https://user-images.githubusercontent.com/1335522/191617582-3707e842-1467-4769-b4b3-44bc862ffd6d.png">

Reply:
There are flags to control whether figures are closed. We got it working in the beam pages.

Reply:
One problem is that any plot code block containing scipy fails due to:

`ModuleNotFoundError: No module named 'scipy'`

(A simple plot block like `plt.plot([11, 12, 13], [16, 15, 14])` renders a graph image fine if I comment out the scipy import; numpy and matplotlib imports work fine.) Is there some way to make a package available inside a plot block? Perhaps this [configuration option](https://matplotlib.org/3.1.3/devel/plot_directive.html#configuration-options):

> plot_working_directory
>     By default, the working directory will be changed to the directory of the example, so the code can get at its data files, if any. Also its path will be added to [sys.path](https://docs.python.org/3/library/sys.html#sys.path) so it can import any helper modules sitting beside it. This configuration option can be used to specify a central directory (also added to [sys.path](https://docs.python.org/3/library/sys.html#sys.path)) where data files and helper modules for all code are located.

Reply:
> There are flags to control whether figures are closed. We got it working in the beam pages.

Thanks--yes, by adding `:context: close-figs` to each plot directive as in the [beam module](https://github.com/sympy/sympy/blob/26f7bdbe3f860e7b4492e102edec2d6b429b5aaf/sympy/physics/continuum_mechanics/beam.py#L1314), different figures can be created. Probably not needed for this page because I plan on including only one plot; good to know how to do.

Reply:
Turned out that SciPy needed to be a dependency for the docs, by adding `scipy` to `doc/requirements.txt`.